app/controllers/concerns/api/users_controller.rb

```diff
@@ -21,5 +21,24 @@ def show
     def me
       render :show
     end
+
+    def suspend
+      authorize(@user, :toggle_suspension_status?)
+
+      target_user = User.find(params[:id])
+      suspend_params = { note_for_current_role: params[:note], user_status: "Suspended" }
```

Comment:
[Question] Based on our call with @pkfrank, do you plan to add information to the note that this user was suspended via API in this PR, or a future PR?

Reply:
I do rather handle that in a separate PR to avoid adding complexity to the review of this PR, in hopes of merging and testing the endpoint in production earlier rather than later.

With some more time to think this through I think we can use the existing [AuditLog](https://github.com/forem/forem/blob/main/app/models/audit_log.rb#L1) model instead of using text in Notes. That class has a `jsonb` data field that we can use to encode the action that was taken much easier than plain text. One example of where this is used is [in the `Organizations::DeleteWorker`](https://github.com/forem/forem/blob/main/app/workers/organizations/delete_worker.rb#L32).

I'll have that follow up PR reference #18121  🙂